frontend/src/common/hooks/useCategoryFilter.ts

```diff
@@ -504,54 +718,53 @@ function summarizeCategories<T extends Categories>(
     // Apply the filters on the original result set
     const rows = applyFilters(originalRows, query.filters);
 
-    // Count the category value occurrences in each category that shares this filter.
+    // Count the category value occurrences in each category that shares this filter. Range categories are not
+    // summarized; they always use the full range of the original result set.
     query.categoryKeys.forEach((categoryKey: CategoryKey) => {
-      accum[categoryKey] = summarizeCategory(categoryKey, rows);
+      if (!isCategoryTypeBetween(categoryKey)) {
+        accum[categoryKey] = summarizeSelectCategory(categoryKey, rows);
+      }
     });
-
     return accum;
   }, {} as FilterState);
 }
 
 /**
- * Count occurrences of category values across the result set for the given category.
+ * Count occurrences of category values across the result set for the given single or multiselect category.
  * @param categoryKey - Category to count category values.
  * @param filteredRows - Array of rows containing category values to count.
  * @return Map of category values keyed by category value key.
  */
-function summarizeCategory<T extends Categories>(
+function summarizeSelectCategory<T extends Categories>(
   categoryKey: CategoryKey,
   filteredRows: Row<T>[]
-): Map<CategoryValueKey, CategoryValue> {
+): KeyedSelectCategoryValue {
   // Aggregate category value counts for each row.
-  return filteredRows.reduce(
-    (accum: Map<CategoryValueKey, CategoryValue>, row: Row<T>) => {
-      // Grab the values of the category for this dataset row.
-      let categoryValues = row.values[categoryKey];
+  return filteredRows.reduce((accum: KeyedSelectCategoryValue, row: Row<T>) => {
+    // Grab the values of the category for this dataset row.
+    let categoryValues = row.values[categoryKey];
 
-      if (!Array.isArray(categoryValues)) {
-        categoryValues = [categoryValues];
-      }
+    if (!Array.isArray(categoryValues)) {
+      categoryValues = [categoryValues];
+    }
 
-      // Init category value it doesn't already exist. Default selected state to false (selected state is updated
-      // from the filter state at a later point).
-      categoryValues.forEach((categoryValueKey: CategoryValueKey) => {
-        let categoryValue = accum.get(categoryValueKey);
-        if (!categoryValue) {
-          categoryValue = {
-            count: 0,
-            key: categoryValueKey,
-            selected: false,
-          };
-          accum.set(categoryValueKey, categoryValue);
-        }
-        // Increment category value count.
-        categoryValue.count++;
-      });
-      return accum;
-    },
-    new Map<CategoryValueKey, CategoryValue>()
-  );
+    // Init category value it doesn't already exist. Default selected state to false (selected state is updated
```

Comment:
Potential addition 😆 ?
```suggestion
    // Init category value if it doesn't already exist. Default selected state to false (selected state is updated
```